scripts/install/install-dependencies.sh

```diff
@@ -253,223 +587,288 @@ function install_latest_cfitsio() {
   mkdir -p "${PANDIR}/astrometry/data"
   python "${POCS}/pocs/utils/data.py"
 }
 
 #-------------------------------------------------------------------------------
+# Do what the user asked us to do...
 
-function show_help() {
-  echo "${THIS_PROGRAM} - Install software needed for PANOPTES."
-  echo " "
-  echo "${THIS_PROGRAM} [options]"
-  echo " "
-  echo "options:"
-  echo "-h, --help                 show brief help"
-  echo "-x                         turn on bash debug output"
-  echo "--run <function>           run the named function and exit, for debugging"
-  echo "--no-apt-get               don't run apt-get to install Linux packages"
-  echo "--no-mongodb               don't install and start mongodb server"
-  echo "--no-cfitsio               don't install the latest version of cfitsio"
-  echo "--no-conda                 don't install the latest version of Anaconda"
-  echo "--no-conda-packages        don't install packages into Anaconda"
-  echo "--rebuild-conda-env        rebuild the panoptes-env"
-  echo "--no-astrometry            don't install astrometry.net software"
-  echo "--no-astrometry-indices    don't install astrometry.net indices"
-  echo "--no-pip-requirements      don't install python packages"
-}
-
+mkdir -p "${PANDIR}"
 cd "${PANDIR}"
 mkdir -p tmp
+mkdir -p "${INSTALL_LOGS_DIR}"
 
-while test ${#} -gt 0; do
-  case "${1}" in
-    -h|--help)
-      show_help
-      exit 0
-      ;;
-    -x)
-      set -x
-      shift
-      ;;
-    --run)
-      if [ $# -lt 2 ]; then
-        echo "Function name missing after ${1}."
-        show_help
-        exit 1
-      fi
-      shift
-      (${1})
-      exit
-      ;;
-    --no-apt-get)
-      DO_APT_GET=0
-      shift
-      ;;
-    --no-mongodb)
-      DO_MONGODB=0
-      shift
-      ;;
-    --no-cfitsio)
-      DO_CFITSIO=0
-      shift
-      ;;
-    --no-conda)
-      DO_CONDA=0
-      shift
-      ;;
-    --rebuild-conda-env)
-      DO_REBUILD_CONDA_ENV=1
-      shift
-      ;;
-    --no-conda-packages)
-      DO_INSTALL_CONDA_PACKAGES=0
-      shift
-      ;;
-    --no-pip-requirements)
-      DO_PIP_REQUIREMENTS=0
-      shift
-      ;;
-    --no-astrometry)
-      DO_ASTROMETRY=0
-      shift
-      ;;
-    --no-astrometry-ind*)
-      DO_ASTROMETRY_INDICES=0
-      shift
-      ;;
-    *)
-      echo "Unknown parameter: ${1}"
-      echo
-      show_help
-      exit 1
-      ;;
-  esac
-done
-
+# For testing, run the named function in a sub-shell, then exit.
+if [[ -n "${DO_RUN_ONE_FUNCTION}" ]] ; then
+  if ("${DO_RUN_ONE_FUNCTION}") ; then
+    echo "Function was successful."
+  else
+    echo "Function failed with status $?"
+  fi
+  exit
+fi
 
+# Install packages using the APT package manager. Works on Debian based systems
+# such as Ubuntu.
 if [[ "${DO_APT_GET}" -eq 1 ]] ; then
   install_apt_packages
 fi
 
-
+# Install Mongodb, a document database. Used by POCS for storing environment
+# readings, from which weather plots are generated.
```

Comment:
Also stores observations metadata. Not really important but comment undersells how much we use it to some extent.